extensions/interactions/base_test.py

```diff
@@ -279,59 +282,44 @@ def test_html_field_types_to_rule_specs_mapping_are_valid(self):
                 html_field_types_to_rule_specs_dict.items()):
             self.assertTrue('format' in html_type_dict)
             self.assertTrue(
-                html_type_dict['format'] in allowed_html_type_formats)
+                html_type_dict['format'] in
+                feconf.ALLOWED_HTML_RULE_VARIABLE_FORMATS)
             del html_type_dict['format']
 
+        html_types = []
         for interaction_id, interaction_rules in (
                 rule_descriptions_dict.items()):
             for rule_type, rule_description in interaction_rules.items():
                 description = rule_description['description']
-                # Extract the html_types from the rule description.
-                html_types_in_description = (
-                    re.findall(r'{{[a-z]\|([^}]*)}', description))
-                for html_type in html_types_in_description:
-                    if 'Html' in html_type and html_type not in html_types:
-                        html_types.append(html_type)
-
+                # Extract the input variables and html_types from the rule
+                # description.
+                input_variables_with_html_type = (
+                    re.findall(r'{{([a-z])\|([^}]*)}', description))
+                input_variables = []
+                for value in input_variables_with_html_type:
+                    if 'Html' in value[1] and value[1] not in html_types:
+                        html_types.append(value[1])
+                    # Check that the input variable belongs to a rule
+                    # spec having html in it.
+                    if value[1] in html_types:
```

Comment:
I don't understand this. Why wouldn't it be? I mean, if it's not in there, you've just put it in, right? (in the previous lines)

Reply:
I had this check because, previously the html_types was global and we needed this.
Removed it now